src/settings-ui/Settings.UI.Library/AwakeProperties.cs

```diff
@@ -15,7 +15,7 @@ public AwakeProperties()
             KeepDisplayOn = false;
             Mode = AwakeMode.PASSIVE;
             IntervalHours = 0;
-            IntervalMinutes = 0;
+            IntervalMinutes = 1;
             ExpirationDateTime = DateTimeOffset.Now;
```

Comment:
Don't we need `Now + 1 Minute`?

Reply:
In timed mode we are calculating duration

```
case AwakeMode.TIMED:
    {
        uint computedTime = (settings.Properties.IntervalHours * 60 * 60) + (settings.Properties.IntervalMinutes * 60);
        SetupTimedKeepAwake(computedTime, settings.Properties.KeepDisplayOn);

        break;
    }
```
We used computedTime in SetTimedKeepAwake

```
Observable.Timer(TimeSpan.FromSeconds(seconds)).Subscribe(
            _ =>
            {
                Logger.LogInfo($"Completed timed thread.");
                CancelExistingThread();
                ResetAwakeMode();
            },
```
"Observable.Timer(awakeDuration).Subscribe" is begining from now and TimeSpan.FromSeconds is giving closest TimeSpan.

The code works as before. The only difference is in the old code when IntervalMinutes was 0, it did not actually work and no effect was observed. Now the time passed when it is 0 is reset immediately. To prevent this, this value was changed to 1.